rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/CharacteristicLongWriteOperation.java

```diff
@@ -209,4 +213,48 @@ public Boolean call(Object emission) {
             }
         };
     }
+
+    private static Func1<Observable<? extends Throwable>, Observable<?>> retryOperationStrategy(
+            final WriteOperationRetryStrategy writeOperationRetryStrategy,
+            final ByteBuffer byteBuffer,
+            final int batchSize) {
+        return new Func1<Observable<? extends Throwable>, Observable<?>>() {
+            @Override
+            public Observable<?> call(Observable<? extends Throwable> observable) {
+                return observable
+                        .flatMap(applyRetryStrategy())
+                        .map(replaceByteBufferPositionForRetry());
+            }
+
+            @NonNull
+            private Func1<Throwable, Observable<WriteOperationRetryStrategy.LongWriteFailure>> applyRetryStrategy() {
+                return new Func1<Throwable, Observable<WriteOperationRetryStrategy.LongWriteFailure>>() {
+                    @Override
+                    public Observable<WriteOperationRetryStrategy.LongWriteFailure> call(Throwable cause) {
+                        if (!(cause instanceof BleException)) {
+                            return Observable.error(cause);
+                        }
+
+                        final int failedBatchNumber = (byteBuffer.position() - batchSize) / batchSize;
+                        final WriteOperationRetryStrategy.LongWriteFailure longWriteFailure =
+                                new WriteOperationRetryStrategy.LongWriteFailure(failedBatchNumber, (BleException) cause);
+
+                        return writeOperationRetryStrategy.call(Observable.just(longWriteFailure));
+                    }
+                };
+            }
+
+            @NonNull
+            private Func1<WriteOperationRetryStrategy.LongWriteFailure, Object> replaceByteBufferPositionForRetry() {
```

Comment:
This could be a simple `.onNext()` as it only passes the `longWriteFailure`

Reply:
Ok.